### ISSUE TITLE
Exception ignored in: <bound method BaseEventLoop.__del__ of <_UnixSelectorEventLoop running=False closed=True debug=False>>

### ISSUE THREAD

**Issue by yangyshe** [KimiNewt/pyshark#347]
python version:3.5.2
pyshark:0.4.2.3
Exception was found when using FileCapture, eventloop was not add by FileCapture's arg,
need to manually close eventloop.
for example:
capture = pyshark.FileCapture(file_dir)
if not capture.eventloop.is_closed():
     capture.eventloop.close()

Traceback as follows:
Exception ignored in: <bound method BaseEventLoop.__del__ of <_UnixSelectorEventLoop running=False closed=True debug=False>>
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/base_events.py", line 431, in __del__
  File "/usr/lib/python3.5/asyncio/unix_events.py", line 58, in close
  File "/usr/lib/python3.5/asyncio/unix_events.py", line 139, in remove_signal_handler
  File "/usr/lib/python3.5/signal.py", line 47, in signal
TypeError: signal handler must be signal.SIG_IGN, signal.SIG_DFL, or a callable object

**Comment by kcexn** [comment]
I'd be happy to take a look at this problem. I've experienced it before too and haven't looked too deeply into it. Do you also experience the following error?

```python
import pyshark as ps
cap = ps.FileCapture('/path/to/file.pcap')
for pkt in cap:
    pass
capture.eventloop.close()
```

Traceback:

Exception ignored in: <bound method Capture.__del__ of <FileCapture /home/kevin/Projects/pyshark/tests/caps/capture_test.pcapng>>
Traceback (most recent call last):
  File "/home/kevin/Projects/pyshark/src/pyshark/capture/capture.py", line 415, in __del__
  File "/home/kevin/Projects/pyshark/src/pyshark/capture/capture.py", line 406, in close
  File "/usr/lib/python3.5/asyncio/base_events.py", line 443, in run_until_complete
  File "/usr/lib/python3.5/asyncio/base_events.py", line 357, in _check_closed
RuntimeError: Event loop is closed
sys:1: RuntimeWarning: coroutine 'Capture._close_async' was never awaited

**Comment by yangyshe** [comment]
It seems not the same question with me, as your traceback indicated that eventloop is closed before closing capture. Follow is the close function in capture.py.
def close(self):
      self.eventloop.run_until_complete(self._close_async())
My issue caused by deleleting eventloop error, that mean eventloop was not closed actually.

**Comment by yangyshe** [comment]
> I'd be happy to take a look at this problem. I've experienced it before too and haven't looked too deeply into it. Do you also experience the following error?
> 
> ```python
> import pyshark as ps
> cap = ps.FileCapture('/path/to/file.pcap')
> for pkt in cap:
>     pass
> cap.eventloop.close()
> ```
> 
> Traceback:
> 
> Exception ignored in: <bound method Capture.**del** of <FileCapture /path/to/file.pcap>>
> Traceback (most recent call last):
> File "/path/to/pyshark/capture/capture.py", line 415, in **del**
> File "/path/to/pyshark/capture/capture.py", line 406, in close
> File "/usr/lib/python3.5/asyncio/base_events.py", line 443, in run_until_complete
> File "/usr/lib/python3.5/asyncio/base_events.py", line 357, in _check_closed
> RuntimeError: Event loop is closed
> sys:1: RuntimeWarning: coroutine 'Capture._close_async' was never awaited

It seems not the same question with me, as your traceback indicated that eventloop is closed before closing capture. Follow is the close function in capture.py.
def close(self):
self.eventloop.run_until_complete(self._close_async())
My issue caused by deleleting eventloop error, that mean eventloop was not closed actually.

**Comment by OXERY** [comment]
> Traceback:
> 
> Exception ignored in: <bound method Capture.**del** of <FileCapture /path/to/file.pcap>>
> Traceback (most recent call last):
> File "/path/to/pyshark/capture/capture.py", line 415, in **del**
> File "/path/to/pyshark/capture/capture.py", line 406, in close
> File "/usr/lib/python3.5/asyncio/base_events.py", line 443, in run_until_complete
> File "/usr/lib/python3.5/asyncio/base_events.py", line 357, in _check_closed
> RuntimeError: Event loop is closed
> sys:1: RuntimeWarning: coroutine 'Capture._close_async' was never awaited

Got the same problem.. It works but spams my console...

**Comment by OXERY** [comment]
after writing `capture.close()` at the end of my function, this error disappeared. ;-)

**Comment by kcexn** [comment]
So this seems to be a known issue with Asyncio in Python 3.5 and is related to the order in which the python garbage collector cleans up resources. The signal handling resources are cleaned up before __del__ is called on the asyncio event loop. This seems to be fixed in Python 3.6 and newer (or at least according to [https://bugs.python.org/issue23548](https://bugs.python.org/issue23548)) It's been fixed since at least Python 3.6.5 and my build of python 3.6.9rc1 also doesn't experience this issue. 

The issue has been raised several times within the python community that I can find:

[https://bugs.python.org/issue23548](https://bugs.python.org/issue23548)
[https://bugs.python.org/issue28628](https://bugs.python.org/issue28628)
[https://github.com/python/asyncio/issues/396](https://github.com/python/asyncio/issues/396)

So the real fix to this is to update to Python 3.6 since this is a problem with the python interpreter. 
However I will submit a fix to put explicit eventloop closing into the FileCapture class so that this issue can be worked around. As well as a context manager to manage this.
